packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { ForceSourceDeployErrorResult } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export function handleDiagnosticErrors(
+  errors: ForceSourceDeployErrorResult,
+  workspacePath: string,
+  sourcePath: string,
+  errorCollection: vscode.DiagnosticCollection
+): vscode.DiagnosticCollection {
+  errorCollection.clear();
+  const diagnosticMap: Map<string, vscode.Diagnostic[]> = new Map();
+  if (errors.hasOwnProperty('result')) {
+    errors.result.forEach(error => {
+      // source:deploy sometimes returns N/A as filePath
+      const fileUri =
+        error.filePath === 'N/A'
```

Comment:
Not all deploy errors have a valid filePath, which maps to your project, so we do the best we can and map to manifest or sourcePath provided for running the deploy command.